docs-ui/components/sample.tsx

```diff
@@ -1,12 +1,89 @@
+import {createContext, ReactChild, useState} from 'react';
+import {ThemeProvider, useTheme} from '@emotion/react';
 import styled from '@emotion/styled';
 
+import {IconMoon} from 'app/icons';
 import space from 'app/styles/space';
+import {darkTheme, lightTheme, Theme} from 'app/utils/theme';
 
-const Sample = styled('div')`
+type ThemeName = 'dark' | 'light';
+
+type Props = {
+  children?: ReactChild;
+  /**
+   * Show the theme switcher, which allows for
+   * switching the local theme context between
+   * light and dark mode. Useful for previewing
+   * components in both modes.
+   */
+  showThemeSwitcher?: boolean;
+  /** Remove the outer border and padding */
+  noBorder?: boolean;
+};
+
+/** Expose the selected theme to children of <Sample /> */
+export const SampleThemeContext = createContext<ThemeName>('light');
+
+const Sample = ({children, showThemeSwitcher = false, noBorder = false}: Props) => {
+  const [themeName, setThemeName] = useState<ThemeName>('light');
+
+  /**
+   * If theme switcher is shown, use the correct theme object based on themeName.
+   * Else, fall back to the global theme object.
+   */
+  const [theme, setTheme] = useState<Theme>(
+    showThemeSwitcher ? (themeName === 'light' ? lightTheme : darkTheme) : useTheme()
+  );
+
+  const toggleTheme = () => {
+    if (themeName === 'light') {
+      setThemeName('dark');
+      setTheme(darkTheme);
```

Comment:
wonder if we should have the current theme name in theme itself... (I was avoiding it so that we wouldn't have logic based on theme name, and instead push people towards using theme aliases).

Reply:
thought of this myself! Would definitely make this cleaner, but wasn't sure if it's worth changing theme.tsx for it.